packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart

```diff
@@ -165,14 +174,80 @@ class _TimelineAnalysisContainerState extends State<TimelineAnalysisContainer>
     );
   }
 
-  Widget _buildProcessingInfo() {
-    return ProcessingInfo(
-      progressValue: widget.processingProgress,
-      processedObject: 'timeline trace',
+  DevToolsTab _buildTab({required String tabName, Widget? trailing}) {
+    return DevToolsTab.create(
+      tabName: tabName,
+      gaPrefix: _gaPrefix,
+      trailing: trailing,
     );
   }
 }
 
+class TimelineEventsView extends StatelessWidget {
+  const TimelineEventsView({
+    Key? key,
+    required this.controller,
+    required this.processing,
+    required this.processingProgress,
+  });
+
+  @visibleForTesting
+  static const emptyTimelineKey = Key('Empty Timeline');
+
+  final PerformanceController controller;
+
+  final bool processing;
+
+  final double processingProgress;
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
the code in this build method is not new. It was pulled out from the above widget.